ui/apps/platform/cypress/helpers/request.js

```diff
@@ -93,3 +94,24 @@ export function interactAndWaitForResponses(
 
     return waitForResponses(routeMatcherMap, waitOptions);
 }
+
+/**
+ * Intercepts a GraphQL request during an interaction and yields the
+ * variables object passed to the request's query
+ *
+ * @param {() => void} interactionCallback The interaction performed to trigger the GraphQL request
+ * @param {string} opname The GraphQL operation name
+ * @returns {Cypress.Chainable<Interception>}
+ */
+export function interactAndInspectGraphQLVariables(interactionCallback, opname) {
+    const key = 'graphql';
```

Comment:
This assumes only one interception for inspection per test?

Asking for understanding, not implying premature generalization.

Reply:
It assumes only one interception per call of `interactAndInspectGraphQLVariables`. I was aiming for a clean slate between multiple calls in a single test to avoid any issue that multiple unbounded interceptors triggering on the same request URL might cause.

There is a bit of repetition in the test with the same opname used many times though. I would like a more concise and clear way to do the same, but nothing else was as easily readable to me. I'll add this to the slow-cooker and see if another approach would be better.

Reply:
Aha, a small change to align with route matcher pattern: replace fixed `key` with `opname` as alias.

Reply:
Oof - good catch 😅